### PR TITLE
Adding unit tests to Protocol component

### PR DESCRIPTION
**Pull request by DukeDavis12**
Adding unit tests to the Protocol component.

Signed-off-by: Davis Benny <davis.benny@intel.com>

### CODE REVIEW

**Review by DukeDavis12 — COMMENTED** [review]
Code coverage is at 70% now.

File: protocol/src/test/java/org/fido/iot/protocol/RendezvousInfoDecoderTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public class RendezvousInfoDecoderTest {
+
+  @Test
+  void getHttpDirectivesTest() throws Exception {
+
+    String ownerdirective  = "http://localhost:8040?ipaddress=10.66.244.99&ownerport=8443";
```
Comment (Darshini-Parikh93) [inline protocol/src/test/java/org/fido/iot/protocol/RendezvousInfoDecoderTest.java:14]: Please remove the IP and probably, you can add a comment that the localhost dnsname or localhost IP address can be replaced by any other IP

File: protocol/src/test/java/org/fido/iot/protocol/RendezvousInfoDecoderTest.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    Composite rvi = RendezvousInfoDecoder.decode(ownerdirective);
+    List<String> ownerDir = RendezvousInfoDecoder.getHttpDirectives(rvi, Const.RV_OWNER_ONLY);
+    assertTrue(ownerDir.get(0).endsWith(":8443"));
+
+    String devdirective  = "https://localhost:8040?ipaddress=10.66.244.99&ownerport=8443";
```
Comment (Darshini-Parikh93) [inline protocol/src/test/java/org/fido/iot/protocol/RendezvousInfoDecoderTest.java:19]: Same as above

File: protocol/src/test/java/org/fido/iot/protocol/RendezvousInfoDecoderTest.java (hunk 1 above)
Comment (DukeDavis12) [reply protocol/src/test/java/org/fido/iot/protocol/RendezvousInfoDecoderTest.java]: Updated. Added the comment also.

File: protocol/src/test/java/org/fido/iot/protocol/RendezvousInfoDecoderTest.java (hunk 2 above)
Comment (DukeDavis12) [reply protocol/src/test/java/org/fido/iot/protocol/RendezvousInfoDecoderTest.java]: Updated.

**Review by Darshini-Parikh93 — APPROVED** [review]

**Review by johneasterday — APPROVED** [review]

**Pull request merged**